ethexe/db/src/database.rs

```diff
@@ -444,6 +447,60 @@ impl Database {
         self.cas.write(data)
     }
 
+    pub fn get_offchain_transaction(&self, tx_hash: H256) -> Option<SignedOffchainTransaction> {
+        self.kv
+            .get(&KeyPrefix::Transaction.one(tx_hash))
+            .map(|data| {
+                Decode::decode(&mut data.as_slice())
+                    .expect("failed to data into `SignedTransaction`")
+            })
+    }
+
+    pub fn set_offchain_transaction(&self, tx: SignedOffchainTransaction) {
+        let tx_hash = tx.tx_hash();
+        self.kv
+            .put(&KeyPrefix::Transaction.one(tx_hash), tx.encode());
+    }
+
+    pub fn check_within_recent_blocks(&self, reference_block_hash: H256) -> Result<()> {
```

Comment:
I'd still separate here cases of Err and "false" by returning Result<bool> giving much more space and flexibility in the end code using this api

Reply:
done